### PR TITLE
BaSP-MR-09-12: AdminUpdate

### PR DESCRIPTION
**Pull request by enriqueanastasini**
The put method to update ADMIN data has been implemented, and the schema has been updated to change the ADMIN data

### CODE REVIEW

**Comment by NicoCagnina** [comment]
you should add a pr description

File: src/models/Admins.js [hunk 1]
```diff
@@ -6,13 +6,13 @@ const adminsSchema = new Schema({
   firstName: {
     type: String,
     required: true,
-    minLength: 3,
+    minLength: 5,
```
Comment (NicoCagnina) [inline src/models/Admins.js:9]: i think this should be a lower value. maybe 3

File: src/models/Admins.js [hunk 2]
```diff
@@ -6,13 +6,13 @@ const adminsSchema = new Schema({
...
     maxLength: 15,
     match: /^[A-Za-z]+$/,
   },
   lastName: {
     type: String,
-    minLength: 3,
+    minLength: 5,
```
Comment (NicoCagnina) [inline src/models/Admins.js:15]: i think this should be a lower value. maybe 3

File: src/validations/admins.js [hunk 3]
```diff
@@ -1,5 +1,40 @@
...
+    dni: Joi.number()
+      .positive()
+      .min(10000000)
+      .max(99999999),
+    phone: Joi.number()
+      .positive()
+      .min(1000000000)
+      .max(9999999999),
+    email: Joi.string()
+      .pattern(/^[^@]+@[^@]+.[a-zA-Z]{2,}$/),
```
Comment (NicoCagnina) [inline src/validations/admins.js:20]: you can avoid the .pattern and simply put .email()

**Review by NicoCagnina — CHANGES_REQUESTED** [review]

**Comment by enriqueanastasini** [comment]
The put method to update ADMIN data has been implemented, and the schema has been updated to change the ADMIN data

File: src/validations/admins.js [hunk 4]
```diff
@@ -1,5 +1,40 @@
...
+      .min(3)
+      .max(15),
+    lastName: Joi.string()
+      .min(3)
+      .max(15),
+    dni: Joi.number()
+      .positive()
+      .min(10000000)
+      .max(99999999),
+    phone: Joi.number()
```
Comment (JuanMornik) [inline src/validations/admins.js:15]: Phone must be a string. Keep in mind that min() and max() value will be changed with string length (9 and 12) and you have to delete positive()

File: src/validations/admins.js [hunk 5]
```diff
@@ -1,5 +1,40 @@
...
+      .max(9999999999),
+    email: Joi.string()
+      .email(),
+    city: Joi.string()
+      .min(2)
+      .max(10),
+    password: Joi.string()
+      .min(8)
+      .pattern(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/),
+  }).min(1);
```
Comment (JuanMornik) [inline src/validations/admins.js:27]: Delete this .min()

**Review by JuanMornik — CHANGES_REQUESTED** [review]

**Review by JuanMornik — APPROVED** [review]

**Review by NicoCagnina — APPROVED** [review]

**Pull request merged**